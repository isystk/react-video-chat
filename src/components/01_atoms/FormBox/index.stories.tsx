import { Meta, Story } from '@storybook/react'
import React from 'react'
import FormBox from './index'

export default {
  title: '01_atoms/FormBox',
  component: FormBox,
} as Meta

const Template: Story = (props) => {
  return <FormBox title="sample" />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
