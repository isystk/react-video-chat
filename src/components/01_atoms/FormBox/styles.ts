import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles((theme) => ({
  breadcrumbs: {
    padding: '20px 0px',
    // listStyle: "none",
    // "& li": {
    //   display: "inline",
    //   listStyle: "none",
    //   fontWeight: "bold",
    //   "&::after": {
    //     content: "'>'",
    //     padding: "0 0.2em",
    //     color: "#555",
    //   },
    //   "&::last-child::after": {
    //     content: "''",
    //   },
    //   "& a": {
    //     textDecoration: "none",
    //     color: "#52b5ee",
    //     "&::hover": {
    //       textDecoration: "underline"
    //     }
    //   }
    // }
  },
  box: {
    backgroundColor: '#fff',
    height: '80vh',
    position: 'relative',
    textAlign: 'center',
  },
  title: {
    backgroundColor: '#3f51b5',
    padding: '10px 20px',
    color: '#fff',
    textAlign: 'left',
  },
  form: {
    margin: '40px auto',
    maxWidth: '500px',
  },
}))

export { useStyles }
