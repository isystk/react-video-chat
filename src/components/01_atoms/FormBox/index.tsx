import React, { FC } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { Box, Breadcrumbs, FormControl, Typography } from '@material-ui/core'
import { connect } from '@/components/hoc'
import { useStyles } from './styles'
import Link from 'next/link'
import { Url } from '@/constants/url'

/** FormBoxProps Props */
export type FormBoxProps = WithChildren & { title }
/** Presenter Props */
export type PresenterProps = FormBoxProps & { classes }

/** Presenter Component */
const FormBoxPresenter: FC<PresenterProps> = ({
  children,
  title,
  classes,
  ...props
}) => (
  <>
    <Breadcrumbs aria-label="breadcrumb" className={classes.breadcrumbs}>
      <Link color="inherit" href={Url.TOP}>
        TOP
      </Link>
      <Typography component="span">{title}</Typography>
    </Breadcrumbs>
    <Box component="div" className={classes.box}>
      <Typography component="div" color="primary" className={classes.title}>
        {title}
      </Typography>
      <FormControl className={classes.form}>{children}</FormControl>
    </Box>
  </>
)

/** Container Component */
const FormBoxContainer: React.FC<
  ContainerProps<FormBoxProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const classes = useStyles()
  return presenter({
    children,
    classes,
    ...props,
  })
}

export default connect<FormBoxProps, PresenterProps>(
  'FormBox',
  FormBoxPresenter,
  FormBoxContainer
)
