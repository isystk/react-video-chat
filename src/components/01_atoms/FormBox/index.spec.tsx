import React from 'react'
import renderer from 'react-test-renderer'
import FormBox from './index'
import '@testing-library/jest-dom/extend-expect'

describe('FormBox', () => {
  it('Match Snapshot', () => {
    const component = renderer.create(<FormBox title="sample" />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
