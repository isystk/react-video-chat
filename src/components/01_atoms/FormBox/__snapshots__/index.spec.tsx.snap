// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBox Match Snapshot 1`] = `
Array [
  <nav
    aria-label="breadcrumb"
    className="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumbs-1 MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      className="MuiBreadcrumbs-ol"
    >
      <li
        className="MuiBreadcrumbs-li"
      >
        <a
          href="http://xxxxxxxxxx:3000/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          TOP
        </a>
      </li>
      <li
        aria-hidden={true}
        className="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        className="MuiBreadcrumbs-li"
      >
        <span
          className="MuiTypography-root MuiTypography-body1"
        >
          sample
        </span>
      </li>
    </ol>
  </nav>,
  <div
    className="MuiBox-root MuiBox-root-5 makeStyles-box-2"
  >
    <div
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1 MuiTypography-colorPrimary"
    >
      sample
    </div>
    <div
      className="MuiFormControl-root makeStyles-form-4"
    />
  </div>,
]
`;
